packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -0,0 +1,184 @@
+};
+
+const actionSection = (primaryAction, secondaryAction) => {
+  let actionButton = null;
+  let dismissButton = null;
+  if (!primaryAction && !secondaryAction) {
+    return null;
+  }
+  if (primaryAction) {
+    actionButton = <Button text={primaryAction.text} variant={Button.Opts.Variants.EMPHASIS} onClick={primaryAction.onClick} />;
+  }
+  if (secondaryAction) {
+    dismissButton = <Button text={secondaryAction.text} onClick={secondaryAction.onClick} />;
+  }
+
+  return <div className={cx('actions')}>{actionButton}{dismissButton}</div>;
+};
+
+const getIcon = (variant, customIcon = null) => {
+  switch (variant) {
+    case variants.ALERT:
+      return (<IconAlert />);
+    case variants.ERROR:
+      return (<IconError />);
+    case variants.WARNING:
+      return (<IconWarning />);
+    case variants.INFO:
+      return (<IconInformation />);
+    case variants.SUCCESS:
+      return (<IconSuccess />);
+    case variants.CUSTOM:
+      return customIcon;
+    default:
+      return null;
+  }
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class NotificationDialog extends React.Component {
+  render() {
+    if (!this.props.isOpen) {
+      return null;
+    }
+
+    const {
+      header,
+      title,
+      message,
+      primaryAction,
+      secondaryAction,
+      variant,
+      customIcon,
+      isOpen,
+      onRequestClose,
+      ...customProps
+    } = this.props;
+
+    const { intl } = this.context;
+
+    const defaultHeader = variant === variants.CUSTOM ? '' : intl.formatMessage({ id: `Terra.notification.dialog.${variant}` });
+    const notificationDialogClassNames = cx('notification-dialog', customProps.className);
+
+    return (
+      <AbstractModal
+        ariaLabel="NotificationDialog"
+        classNameModal={notificationDialogClassNames}
+        isOpen={this.props.isOpen}
+        onRequestClose={this.props.onRequestClose}
+        closeOnEsc={false}
+        closeOnOutsideClick={false}
+        zIndex="8000"
+      >
+        <div className={cx('header-body')}>{header || defaultHeader}</div>
+        <div className={cx('notification-dialog-body')}>
+          {variant &&
+            <div className={cx('icon-div')}>{getIcon(variant, customIcon)}</div>
+          }
+          <div className={cx('notification-dialog-body-text')}>
+            <div className={cx('title')}>{title}</div>
+            <div className={cx('message')}>{message}</div>
+          </div>
+        </div>
+        <div className={cx('footer-body')}>{actionSection(primaryAction, secondaryAction)}</div>
+      </AbstractModal>
+    );
+  }
+}
+
+NotificationDialog.propTypes = propTypes;
+NotificationDialog.defaultProps = defaultProps;
+NotificationDialog.contextTypes = contextTypes;
+NotificationDialog.Opts = {};
```

Comment:
Instead of attaching Opts and Variants to NotificationDialog, we've landed exporting them as a separate export from the component file. This can be replaced with the following code:
```
export {  NotificationDialogVariants: Variants };
```